webhooks/clusterclass.go

```diff
@@ -161,23 +160,25 @@ func (webhook *ClusterClass) validateCompatibleSpecChanges(old, in *clusterv1.Cl
 	allErrs = append(allErrs, webhook.validateMachineDeploymentsCompatibleChanges(old, in)...)
 
 	// Validate InfrastructureClusterTemplate changes in a compatible way.
-	allErrs = append(allErrs, webhook.validateTemplatesAreCompatible(in.Spec.Infrastructure,
-		old.Spec.Infrastructure,
-		field.NewPath("spec", "infrastructure"),
-	)...)
+	if err := check.TemplatesAreCompatible(in.Spec.Infrastructure, old.Spec.Infrastructure); err != nil {
```

Comment:
I think this would be nicely simplified if `TemplatesAreCompatible` just returns a field.Error (pending discussion on the other conversation)

Reply:
I've got everything returning field.ErrorList now - I think that covers most of the use cases at the cost of changing the API here, but it's low impact.